### ISSUE TITLE
[Bug] Crashing creating world "IllegalStateException: Feature order cycle found, involved sources:" when trying to add Terralith+Tectonic in modpack and generating world

### ISSUE THREAD

**Issue by hiimrickross** [Stardust-Labs-MC/Terralith#114]
### What Minecraft version(s) does this apply to?

1.20

### Project version

1.20_v2.5.0

### Mod or datapack version?

Mod (Forge)

### Describe the bug

Hi, 
I've been seeing SO many videos recently about Terralith, Tectonic, and Distant Horizons on youtube over these last few months, so I wanted to try it out on my 1.20.1 instance of Forge Prominence Classic v2.7.0. I know I need to start a new world for it, however when I do it crashes blaming terralith:lavender_forest biome and terralith:blooming_valley specifically, with java.langIllegalStateException. It's probably more than just this biome, but yes it doesn't seem to like something with these two.

I can't make heads of what exactly is crashing in this error, but I do have the crash log that appeared (and normal log)

### Additional context

https://gist.github.com/hiimrickross/3a8445c36b30d2b323f9285052cc0b25 - crash report
https://gist.github.com/hiimrickross/98cf50a423993f3d4219e44ec2850eb1 - log report

**Comment by catter1** [comment]
This is an [issue with Croptopia](https://github.com/ExcessiveAmountsOfZombies/Croptopia/issues/577) that was fixed about 5 months ago. Update to at least v3.0.4 and you should be all set.

**Comment by hiimrickross** [comment]
> This is an [issue with Croptopia](https://github.com/ExcessiveAmountsOfZombies/Croptopia/issues/577) that was fixed about 5 months ago. Update to at least v3.0.4 and you should be all set.

Amazing, thank you!
It seems to have worked..
I have added all the latest versions of the "mod" type of: Terralith, Tectonic, and Lithostitched (says it's an optional dependency for Tectonic) as well as updated Croptopia to 3.0.4 and now it has allowed me to generate a world. Will have to fly around for a while and check to see if all "seems" to be working correctly. I had tried just now with "only" adding Tectonic as that would work, and the world looked amazing and seemed to generate perfectly.

-edit
Does seem like Terralith is working properly, /locate biome shows lots of terralith entries, and the world does look more "vast" and bigger than when it was simply 'just' Tectonic. It seems like it makes the spacing of structure mods (dungeons arise, enhanced, just various structure mods) much more rare though as the world is now ginormous. It is so nice though, that (either just tectonic or terra+tectonic) seem to fix that modded minecraft experince of chunk gen being so broken very often! The landscape looks so much smoother and consistent, rather than odd blocks of land and floating pieces and such.
